js/src/api/transport/ws/ws.js

```diff
@@ -65,32 +78,101 @@ export default class Ws extends JsonRpcBase {
     this._ws.onopen = this._onOpen;
     this._ws.onclose = this._onClose;
     this._ws.onmessage = this._onMessage;
+
+    // Get counts in dev mode
+    if (process.env.NODE_ENV === 'development') {
+      this._count = 0;
+      this._lastCount = {
+        timestamp: Date.now(),
+        count: 0
+      };
+
+      window.setInterval(() => {
+        const n = this._count - this._lastCount.count;
+        const t = (Date.now() - this._lastCount.timestamp) / 1000;
+        const s = Math.round(1000 * n / t) / 1000;
+
+        if (this._debug) {
+          console.log('::parityWS', `speed: ${s} req/s`, `count: ${this._count}`);
+        }
+      }, 5000);
+
+      window._parityWS = this;
+    }
+  }
+
+  _checkNodeUp () {
+    const url = process.env.PARITY_URL || window.location.host;
+
+    return fetch(
+        `http://${url}/api/ping`,
+        { method: 'HEAD' }
+      )
+      .then((r) => {
+        return r.status === 200;
+      }, () => {
+        return false;
+      })
+      .catch(() => false);
   }
 
   _onOpen = (event) => {
     console.log('ws:onOpen', event);
     this._connected = true;
     this._connecting = false;
     this._autoConnect = true;
+    this._retries = 0;
 
     Object.keys(this._messages)
       .filter((id) => this._messages[id].queued)
       .forEach(this._send);
   }
 
   _onClose = (event) => {
-    console.log('ws:onClose', event);
     this._connected = false;
     this._connecting = false;
 
-    if (this._autoConnect) {
-      setTimeout(() => this._connect(), 500);
-    }
+    this._checkNodeUp()
+      .then((up) => {
+        // If the connection has been closed and the node
+        // is up, it means we have a wrong token
+        // Event code 1006 for WS means there is an error
+        // (not just closed by server)
+        if (up && event.code === 1006) {
+          event.status = 403;
```

Comment:
Dangerous, but better than what we have atm. (Apart from the up check that doesn't belong)